cachito/workers/tasks/gomod.py

```diff
@@ -48,5 +49,8 @@ def fetch_gomod_source(request_id, dep_replacements=None):
         "GOPATH": {"value": "deps/gomod", "kind": "path"},
     }
     env_vars.update(config.cachito_default_environment_variables.get("gomod", {}))
-    update_request_with_package(request_id, module, env_vars)
-    update_request_with_deps(request_id, module, deps)
+    update_request_with_package(request_id, gomod["module"], env_vars)
+    update_request_with_deps(request_id, gomod["module"], gomod["module_deps"] + [gomod["pkg"]])
```

Comment:
The package of the module doesn't seem like a dependency to me. Since the top-level package and the module have the same name, you can correlate the go module to the go package in the "packages" JSON array without this.

Could you please explain your point of view?